src/Stack/Options.hs

```diff
@@ -279,6 +280,12 @@ configOptsParser hide0 =
            <> help "Extra directories to check for libraries"
            <> hide
             )))
+    <*> optionalFirst (textOption
+             ( long "with-ghc"
```

Comment:
That should probably be `"with-gcc"` – not sure if there would be an advantage in using `configMonoidOverrideGccPathName` directly…